discovery-provider/src/queries/search_es.py

```diff
@@ -184,6 +189,10 @@ def tag_match(fieldname):
             dsl = tag_match("tag_list")
             dsl["query"]["bool"]["must"].append(be_saved(current_user_id))
             mdsl.extend([{"index": ES_TRACKS}, dsl])
+        if exclude_premium:
+            mdsl.extend(
+                [{"index": ES_TRACKS}, {"term": {"is_premium": {"value": False}}}]
```

Comment:
Realizing that `is_premium` will have to be added as a field in `TrackIndexer.ts` and we'll have to roll forward to `tracks8` in `indexNames.ts` to effect this change.

ES will silently ignore term queries on unmapped fields I think... which is kind of surprising.  Maybe there's a setting somewhere to make this more strict.  One would hope this query on an unmapped field would return an error.

Reply:
hmm interesting

Reply:
updated! please re-review :)